Add React propTypes?

Turns out that the [**eslint-plugin-react**](https://github.com/yannickcr/eslint-plugin-react#recommended) plugin includes a [set of recommended rules](https://github.com/yannickcr/eslint-plugin-react/blob/fcacb51579154e4acb3da416b805fad4bd1a4b94/index.js#L100-L125):

```json
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
```

```sh
$ npm run lint:js

> snoozetabs@1.0.0 lint:js /Users/pdehaan/dev/github/mozilla/SnoozeTabs
> eslint --max-warnings=0 .


/Users/pdehaan/dev/github/mozilla/SnoozeTabs/src/lib/components/DatePickerPanel.js
   13:27  error  'defaultValue' is missing in props validation  react/prop-types
   23:13  error  'id' is missing in props validation            react/prop-types
   23:17  error  'header' is missing in props validation        react/prop-types
   23:25  error  'active' is missing in props validation        react/prop-types
   23:33  error  'onClose' is missing in props validation       react/prop-types
  106:13  error  'onSelect' is missing in props validation      react/prop-types

/Users/pdehaan/dev/github/mozilla/SnoozeTabs/src/lib/components/MainPanel.js
  19:13  error  'id' is missing in props validation                  react/prop-types
  19:17  error  'active' is missing in props validation              react/prop-types
  19:25  error  'switchPanel' is missing in props validation         react/prop-types
  55:13  error  'scheduleSnoozedTab' is missing in props validation  react/prop-types
  69:13  error  'scheduleSnoozedTab' is missing in props validation  react/prop-types

/Users/pdehaan/dev/github/mozilla/SnoozeTabs/src/lib/components/ManagePanel.js
   20:13  error  'id' is missing in props validation                react/prop-types
   20:17  error  'entries' is missing in props validation           react/prop-types
   20:26  error  'active' is missing in props validation            react/prop-types
   20:34  error  'switchPanel' is missing in props validation       react/prop-types
   20:47  error  'tabIsSnoozable' is missing in props validation    react/prop-types
   96:13  error  'openSnoozedTab' is missing in props validation    react/prop-types
   96:29  error  'cancelSnoozedTab' is missing in props validation  react/prop-types
  121:18  error  'openSnoozedTab' is missing in props validation    react/prop-types
  141:13  error  'updateSnoozedTab' is missing in props validation  react/prop-types

/Users/pdehaan/dev/github/mozilla/SnoozeTabs/src/lib/components/SnoozePopup.js
  8:13  error  'activePanel' is missing in props validation  react/prop-types

✖ 21 problems (21 errors, 0 warnings)


npm ERR! Darwin 16.3.0
npm ERR! argv "/Users/pdehaan/.nvm/versions/node/v6.9.4/bin/node" "/Users/pdehaan/.nvm/versions/node/v6.9.4/bin/npm" "run" "lint:js"
npm ERR! node v6.9.4
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! snoozetabs@1.0.0 lint:js: `eslint --max-warnings=0 .`
npm ERR! Exit status 1
```

Looks like the only rule we're failing is [**react/prop-types**](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md). ([Docs](https://facebook.github.io/react/docs/typechecking-with-proptypes.html))

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -5,7 +5,10 @@
     "es6": true,
     "webextensions": true
   },
-  "extends": "eslint:recommended",
+  "extends": [
+    "eslint:recommended",
+    "plugin:react/recommended"
+  ],
   "plugins": [
     "react",
     "promise"

--- a/src/lib/components/DatePickerPanel.js
+++ b/src/lib/components/DatePickerPanel.js
@@ -144,3 +144,13 @@ export default class DatePickerPanel extends React.Component {
     }
   }
 }
+
+DatePickerPanel.propTypes = {
+  active: React.PropTypes.bool.isRequired,
+  defaultValue: React.PropTypes.string.isRequired,
+  header: React.PropTypes.string.isRequired,
+  id: React.PropTypes.string.isRequired,
+  moment: React.PropTypes.func.isRequired,
+  onClose: React.PropTypes.func.isRequired,
+  onSelect: React.PropTypes.func.isRequired,
+};

--- a/src/lib/components/MainPanel.js
+++ b/src/lib/components/MainPanel.js
@@ -103,3 +103,11 @@ export default class MainPanel extends React.Component {
     scheduleSnoozedTab(dateChoice, PICK_TIME);
   }
 }
+
+MainPanel.propTypes = {
+  active: React.PropTypes.bool.isRequired,
+  id: React.PropTypes.string.isRequired,
+  moment: React.PropTypes.func.isRequired,
+  scheduleSnoozedTab: React.PropTypes.func.isRequired,
+  switchPanel: React.PropTypes.func.isRequired,
+};

--- a/src/lib/components/ManagePanel.js
+++ b/src/lib/components/ManagePanel.js
@@ -199,3 +199,17 @@ export default class ManagePanel extends React.Component {
   }
 
 }
+
+ManagePanel.propTypes = {
+  dontShow: React.PropTypes.bool.isRequired,
+  active: React.PropTypes.bool.isRequired,
+  entries: React.PropTypes.array.isRequired,
+  id: React.PropTypes.string.isRequired,
+  moment: React.PropTypes.func.isRequired,
+  openSnoozedTab: React.PropTypes.func.isRequired,
+  cancelSnoozedTab: React.PropTypes.func.isRequired,
+  updateSnoozedTab: React.PropTypes.func.isRequired,
+  tabIsSnoozable: React.PropTypes.func.isRequired,
+  updateDontShow: React.PropTypes.func.isRequired,
+  switchPanel: React.PropTypes.func.isRequired,
+};

--- a/src/lib/components/SnoozePopup.js
+++ b/src/lib/components/SnoozePopup.js
@@ -83,3 +83,14 @@ export default class SnoozePopup extends React.Component {
     }
   }
 }
+
+SnoozePopup.propTypes = {
+  cancelSnoozedTab: React.PropTypes.func.isRequired,
+  getAlarmsAndProperties: React.PropTypes.func.isRequired,
+  moment: React.PropTypes.func.isRequired,
+  openSnoozedTab: React.PropTypes.func.isRequired,
+  queryTabIsSnoozable: React.PropTypes.func.isRequired,
+  scheduleSnoozedTab: React.PropTypes.func.isRequired,
+  updateDontShow: React.PropTypes.func.isRequired,
+  updateSnoozedTab: React.PropTypes.func.isRequired,
+};

